stats/stats.go

```diff
@@ -0,0 +1,82 @@
+// +build linux
+
+package stats
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"time"
+
+	pb "github.com/c-fs/cfs/proto"
+	"github.com/google/cadvisor/info/v1"
+	"github.com/google/cadvisor/manager"
+	"github.com/google/cadvisor/storage/memory"
+	"github.com/google/cadvisor/utils/sysfs"
+	"golang.org/x/net/context"
+)
+
+const (
+	// TODO: make container name configurable
+	//
+	// The recommended way to run cfs is putting the process in an
+	// exclusive container, which has its own cgroup in each hierarchy.
+	// It helps cfs to monitor its resource usage.
+	//
+	// It asks users to specify container name instead of detecting
+	// cgroup that has the process automatically because it may meet
+	// strange cases:
+	// 1. cfs process may be in different cpu/memory/etc cgroups
+	// 2. the cgroup that includes cfs may have other processes
+	// So it hopes that user could take care of it.
+	DefaultContainerName = "/cfs"
+
+	statsToCacheNum = 60
+	storageDuration = 2 * time.Minute
+)
+
+var globalServer *server
+
+func init() {
+	var err error
+	globalServer, err = newServer(DefaultContainerName)
+	if err != nil {
+		log.Printf("metrics: init DefaultServer: %v", err)
+	}
+}
+
+func Server() pb.StatsServer { return globalServer }
+
+type server struct {
+	cmgr  manager.Manager
+	cname string
+}
+
+func newServer(containerName string) (*server, error) {
+	sysFs, err := sysfs.NewRealSysFs()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a system interface (%v)", err)
+	}
+	// TODO: support influxdb or other backend storage
+	cmgr, err := manager.New(memory.New(storageDuration, nil), sysFs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a container Manager (%v)", err)
+	}
+	if err := cmgr.Start(); err != nil {
+		return nil, fmt.Errorf("failed to start container manager (%v)", err)
+	}
+	return &server{cmgr: cmgr, cname: containerName}, nil
+}
+
+func (s *server) ContainerInfo(ctx context.Context, req *pb.ContainerInfoRequest) (*pb.ContainerInfoReply, error) {
+	infoReq := v1.DefaultContainerInfoRequest()
+	info, err := s.cmgr.GetContainerInfo(s.cname, &infoReq)
+	if err != nil {
+		return nil, err
```

Comment:
return empty proto message. i do not think nil is valid. it will panic the serving routine. try it yourself. thanks.